File: golang/cmd/create_domain/create_domain.go
// Usage: TOKEN=token go run cmd/create_domain/create_domain.go example.com
package main

import (
	"fmt"
	"os"
	"strconv"

	"github.com/dnsimple/dnsimple-go/dnsimple"
)

func main() {
	oauthToken := os.Getenv("TOKEN")
	client := dnsimple.NewClient(dnsimple.NewOauthTokenCredentials(oauthToken))
	client.BaseURL = "https://api.sandbox.dnsimple.com"

	// get the current authenticated account (if you don't know who you are)
	whoamiResponse, err := client.Identity.Whoami()
	if err != nil {
		fmt.Printf("Whoami() returned error: %v\n", err)
		os.Exit(1)
	}

	accountId := strconv.Itoa(whoamiResponse.Data.Account.ID)

	name := os.Args[1]
	fmt.Printf("Adding domain %v\n", name)
	domain := dnsimple.Domain{Name: name}
	createDomainResponse, err := client.Domains.CreateDomain(accountId, domain)
	if err != nil {
		fmt.Printf("CreateDomain() returned error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("%+v\n", createDomainResponse.Data)
}

File: csharp/Examples/CreateDomain.cs
using dnsimple;
using dnsimple.Services;

namespace DnsimpleExamples.Examples;

// Usage: TOKEN=your-token dotnet run --framework net10.0 -- create_domain example.com
public static class CreateDomain
{
    public static void Run(string[] args)
    {
        var token = Environment.GetEnvironmentVariable("TOKEN");
        if (string.IsNullOrEmpty(token))
        {
            Console.Error.WriteLine("The TOKEN environment variable is required");
            Environment.Exit(1);
        }

        if (args.Length < 1)
        {
            Console.Error.WriteLine("A domain name argument is required");
            Environment.Exit(1);
        }

        var name = args[0];

        var client = new Client();
        client.ChangeBaseUrlTo("https://api.sandbox.dnsimple.com");
        client.AddCredentials(new OAuth2Credentials(token));

        // get the current authenticated account (if you don't know who you are)
        var accountId = client.Identity.Whoami().Data.Account.Id;

        Console.WriteLine($"Adding domain {name}");

        // add a domain to the account (this does not register the domain,
        // it simply starts hosting it - see the `check`/`transfer_domain`
        // examples for registering/transferring a domain)
        try
        {
            var domain = client.Domains.CreateDomain(accountId, new Domain { Name = name }).Data;
            Console.WriteLine($"Domain: Id={domain.Id}, Name={domain.Name}, State={domain.State}");
        }
        catch (DnsimpleException e)
        {
            Console.Error.WriteLine($"CreateDomain() returned error: {e.Message}");
            Environment.Exit(1);
        }
    }
}
